### PR TITLE
[WIP] Tweak Development Setup

### PR DESCRIPTION
**Pull request by jtigger**
Minor tweaks to #3251, all aiming to continue to smooth the ride.

- be explicit about versions of dependencies.
- rely on package managers for installing software.
- ...

(this is WIP)

### CODE REVIEW

File: docs/setting-up-local-development.md [hunk 1]
```diff
@@ -2,73 +2,187 @@
...
 
-* Copy the Vagrant Procfile example `cp Procfile_Vagrant.example Procfile_Vagrant`
-* Start the server with: `foreman s -f Procfile_Vagrant` (the `-f` explicitly states which Procfile to use and we don't want to use the original Procfile in our vagrant environments)
-* Sometimes you need to: `bundle exec foreman s -f Procfile_Vagrant`
-* Then you can access the local server at [localhost:3030](http://localhost:3030).
-* You can log in as a test user using the `assume` dropdown menu on the top right of the page without creating any new user for the app.
+With this, we're ready to download, build and run exercism.io locally (skip down to [The Code](#get-the-source-code))...
 
-_Again this is assuming you are forwarding port 3000 to 3030 in your Vagrantfile, adjust accordingly to your environment_
+### Running exercism.io in a Docker container via Vagrant
```
Comment (tejasbubane) [inline docs/setting-up-local-development.md]: Not sure if `vagrant` should be mentioned here. Sure windows requires it, but linux and mac don't. The docker readme (linked below) has specific instructions.

Comment (kytrinyx) [inline docs/setting-up-local-development.md]: I think you're probably right about this one—let's only mention it where it's actually a dependency. Or, we could rearrange things to put dependencies at the top with caveats about where they're needed.

Comment (jtigger) [inline docs/setting-up-local-development.md]: At the start of this "tweak", I was aiming to re-organize the existing instructions (which didn't caveat using vagrant).

Yeah, the Docker readme is really nice... almost just want to point them to that and call it done.

This is really helpful, @tejasbubane!  I haven't used the Docker-based setup and have been kinda "wingin' it" with these here instructions.  I'll find a way to keep the main flow of instructions clear while handling these differences.

Comment (kytrinyx) [inline docs/setting-up-local-development.md]: I'm happy with merging this without the tweaks and accepting a second pull request to improve the docs.

Either way—just give me a heads up when you're ready for a final look.

Comment (jtigger) [inline docs/setting-up-local-development.md]: Thanks, Katrina.  I'm gonna take you up on that.  This PR has been open for quite a while and I'd like to land it.  I'll signal when ready.

**Comment by jtigger** [comment]
I went to rebase and realized that I accidentally created this branch in the `exercism` org's repo, not my fork.  Closing this to do the PR properly.

**Pull request closed without merging**